services/spar/test-integration/Test/Spar/APISpec.hs

```diff
@@ -1305,6 +1307,76 @@ specScimAndSAML = do
 
     liftIO $ mid `shouldBe` Just (ScimT.scimUserId scimStoredUser)
 
+specProvisionScimAndSAMLUserWithRole :: SpecWith TestEnv
+specProvisionScimAndSAMLUserWithRole = do
+  describe "provision scim user with SAML with role" $ do
+    it "create user" $ do
+      (tok, (owner, tid, _idp, (_, _privcreds))) <- ScimT.registerIdPAndScimTokenWithMeta
+      let testCreateUserWithRole role = do
+            scimUser <- do
+              u <- ScimT.randomScimUser
+              pure $ u {Scim.roles = [cs $ toByteString $ role]}
+            userId <- ScimT.scimUserId <$> ScimT.createUser tok scimUser
+            ScimT.checkTeamMembersRole tid owner userId role
+      mapM_ testCreateUserWithRole [minBound .. maxBound]
+    it "create user - default to member if no role given" $ do
+      (tok, (owner, tid, _idp, (_, _privcreds))) <- ScimT.registerIdPAndScimTokenWithMeta
+      scimUser <- do
+        u <- ScimT.randomScimUser
+        pure $ u {Scim.roles = []}
+      userId <- ScimT.scimUserId <$> ScimT.createUser tok scimUser
+      ScimT.checkTeamMembersRole tid owner userId RoleMember
+    it "create user - fail if more than one role given" $ do
+      (tok, _) <- ScimT.registerIdPAndScimTokenWithMeta
+      scimUser <- do
+        u <- ScimT.randomScimUser
+        pure $ u {Scim.roles = ["member", "admin"]}
+      ScimT.createUser' tok scimUser !!! do
+        const 400 === statusCode
+        const (Just "A user cannot have more than one role.") =~= responseBody
+    it "create user - fail if role name cannot be parsed correctly" $ do
+      (tok, _) <- ScimT.registerIdPAndScimTokenWithMeta
+      scimUser <- do
+        u <- ScimT.randomScimUser
+        pure $ u {Scim.roles = ["president"]}
+      ScimT.createUser' tok scimUser !!! do
+        const 400 === statusCode
+        const (Just "The role 'president' is not valid. Valid roles are owner, admin, member, partner.") =~= responseBody
+    it "update user" $ do
+      (tok, (owner, tid, _idp, (_, _privcreds))) <- ScimT.registerIdPAndScimTokenWithMeta
+      scimUserWithDefaultRole <- ScimT.randomScimUser
+      userId <- ScimT.scimUserId <$> ScimT.createUser tok scimUserWithDefaultRole
+      let testUpdateUserWithRole role = do
+            let scimUserWithRole = scimUserWithDefaultRole {Scim.roles = [cs $ toByteString $ role]}
+            _ <- ScimT.updateUser tok userId scimUserWithRole
+            ScimT.checkTeamMembersRole tid owner userId role
+      mapM_ testUpdateUserWithRole [minBound .. maxBound]
+    it "update user - default to member if no role given" $ do
```

Comment:
shouldn't this be the previous value instead of the default?  not sure, though.  i think we have an issue here with not being able to distinguish `[]` from `null` or missing.

Reply:
IMO it is not surprising for the SCIM admin when they change the `roles` field from the previous SCIM provider setting (e.g. `[owner]`) to `[]` or `null` that this leads to the role being reset to the default. What would you expect as a user?

Reply:
ok, makes sense.  maybe we can add something to that end somewhere under docs.wire.com?  (separate PR though.)